### ISSUE TITLE
[TagEdit Subtag Completion] Popup shows only the left edge of the first letter of each tag

### ISSUE THREAD

**Issue by aleksejrs** [glutanimate/anki-addons-misc#14]

**Comment by aleksejrs** [comment]
Can't attach the screenshot, an error happens.

**Comment by aleksejrs** [comment]
![ankitagaddonissue](https://user-images.githubusercontent.com/1025861/29516458-127c6fdc-8679-11e7-99f7-79c96dbe946d.png)

**Comment by glutanimate** [comment]
Thanks for letting me know.

- What OS / Anki versions are you using?
- Do your tags contain any special characters / are they in a non-latin alphabet?
- Are the tags displayed properly if you comment [this line](https://github.com/glutanimate/anki-addons-misc/blob/master/src/tagedit_subtag_completer.py#L163)?

**Comment by aleksejrs** [comment]
* Debian GNU/Linux stable, 2.0.47
* It doesn't matter.
* Doesn't help. It doesn't depend on the LIMIT_TO_HIERARCHY option.
* Doesn't help.

When I activate the field, the full list of tags is displayed correctly.

**Comment by glutanimate** [comment]
Just tested this on most of my installations. Here are the results:

- Windows 10 with Anki binary build – not reproducible
- Ubuntu 12.04 with Anki binary build, GTK theme – not reproducible
- Ubuntu 12.04 with Anki source package, GTK theme – not reproducible
- Kubuntu 14.04 with Anki binary build
    + GTK theme – not reproducible
    + Oxygen theme – reproducible
    + QtCurve theme – reproducible

I can also reliably fix the issue by commenting out the line I mentioned in my previous comment. Please make sure you selected the right line in your tests. This is the only segment of the code that directly modifies the popup element.

In summary, this seems to be a toolkit-related issue that is somehow associated with specific themes and/or theming engines. No clue how to address this, honestly. Seems out of my reach for now. Hopefully it will be solved once 2.1 comes out and I adapt the add-on for it. I might have to make the search term highlighting configurable until then, and it disable it by default.

**Comment by aleksejrs** [comment]
> Please make sure you selected the right line in your tests.

Oops. It was the wrong line, and commenting the right one out helps indeed.  Thanks!
Just changing the theme to supposedly ordinary GTK ones didn't help.

I thought the page was scrolled to the right line, because pressing Enter on the URL did nothing, but there was actually not enough vertical space to scroll it down to that line, and when I follow the link, the page is scrolled to the right a little if the text size is larger than my default (probably to make some icon visible), so that I don't see the marker and sometimes the line numbers. It seemed very strange that I should comment the "else:" out…

**Comment by glutanimate** [comment]
Ok, no worries. GitHub's line linking can be fickle sometimes.

This issue should be fixed with 413f2976784e6840545ba12a5c8d250e5e5f17ce. Please give the latest GitHub version a try and let me know if it works for you. The add-on should now also be compatible with Anki 2.1.x.

**Comment by aleksejrs** [comment]
This issue is fixed.

The 2.1 compatibility causes an error on 2.0.47:

```
  File "/home/fc/.local/share/Anki2/addons/tagedit_subtag_completer.py", line 55, in paint
    if options.widget is None:
AttributeError: 'QStyleOptionViewItem' object has no attribute 'widget'
```

**Comment by glutanimate** [comment]
Ok, the add-on should work on 2.0.x now (f00eadc8e5587e6f267be8f0297a32acc0fe2d99). Thanks for report, once again.
